File: os2skoledata-google-workspace-sync/Services/OS2skoledata/Model/DBUser.cs
using System.Collections.Generic;

namespace os2skoledata_google_workspace_sync.Services.OS2skoledata.Model
{
    public class DBUser
    {
        public long DatabaseId { get; set; }
        public string LocalPersonId { get; set; }
        public string Cpr { get; set; }
        public string Firstname { get; set; }
        public string FamilyName { get; set; }
        public string Username { get; set; }
        public DBRole Role { get; set; }
        public DBRole GlobalRole { get; set; }
        public List<string> GroupIds { get; set; }
        public List<Institution> Institutions { get; set; }
        public Institution CurrentInstitution { get; set; }
        public bool Deleted;
        public List<string> TotalRoles { get; set; }
        public string StilUsername { get; set; }
        public string UniId { get; set; }
        public List<DBContactCard> ContactCards { get; set; }
        public bool SetPasswordOnCreate { get; set; }
        public string Password { get; set; }
        public string ReservedUsername { get; set; }
        public Institution PrimaryInstitution { get; set; }

        // empty for all other than students
        public List<string> StudentMainGroups { get; set; }
        public List<string> StudentMainGroupsGoogleWorkspaceIds { get; set; }
        public List<MiniGroup> StudentMainGroupsAsObjects { get; set; }
        public StudentRole StudentRole { get; set; }
        public StudentRole GlobalStudentRole { get; set; }
        public long StudentMainGroupStartYearForInstitution { get; set; }
        public string StudentMainGroupLevelForInstitution { get; set; }

        // empty for all other than employees
        public List<EmployeeRole> EmployeeRoles { get; set; }
        public List<EmployeeRole> GlobalEmployeeRoles { get; set; }

        // empty for all other than externals
        public ExternalRole ExternalRole { get; set; }
        public ExternalRole GlobalExternalRole { get; set; }

        // not from api, only for keeping track of members of security groups
        public bool IsExcluded {get; set; } = false;
    }

    public enum DBRole
    {
        STUDENT,
        EMPLOYEE,
        EXTERNAL
    }

    public enum StudentRole
    {
        BARN,
        ELEV,
        STUDERENDE,
        UNKNOWN
    }

    public enum EmployeeRole
    {
        LÆRER,
        PÆDAGOG,
        VIKAR,
        LEDER,
        LEDELSE,
        TAP,
        KONSULENT,
        UNKNOWN
    }

    public enum ExternalRole
    {
        PRAKTIKANT,
        EKSTERN,
        UNKNOWN
    }
}

File: os2skoledata-azure-ad-sync/src/main/java/dk/digitalidentity/service/model/DBUser.java
package dk.digitalidentity.service.model;

import dk.digitalidentity.service.model.enums.EmployeeRole;
import dk.digitalidentity.service.model.enums.ExternalRole;
import dk.digitalidentity.service.model.enums.Role;
import dk.digitalidentity.service.model.enums.StudentRole;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class DBUser {
	private long databaseId;
	private String localPersonId;
	private String cpr;
	private String firstName;
	private String familyName;
	private String username;
	private String stilUsername;
	private Role role;
	private Role globalRole;
	private List<String> groupIds;
	private List<Institution> institutions;
	private String currentInstitutionNumber;
	private List<String> studentMainGroups;
	private StudentRole studentRole;
	private StudentRole globalStudentRole;
	private List<EmployeeRole> employeeRoles;
	private List<EmployeeRole> globalEmployeeRoles;
	private ExternalRole externalRole;
	private ExternalRole globalExternalRole;
	private boolean deleted;
	private String uniId;
	private String studentMainGroupLevelForInstitution;
	private boolean setPasswordOnCreate;
	private String password;

	// not from api. Used to securityGroup and team sync in full sync
	private String azureId;
}
